src/meili/map_matcher.cc

```diff
@@ -167,8 +190,9 @@ std::vector<MatchResult> InterpolateMeasurements(const MapMatcher& mapmatcher,
     return results;
   }
 
-  std::vector<EdgeSegment> route = MergeRoute(mapmatcher.state_container().state(stateid),
-                                              mapmatcher.state_container().state(next_stateid));
+  std::vector<EdgeSegment> route;
```

Comment:
This function, as well as many others in this PR, are missing unit tests (new or updated ones). Are we planning on adding those?

Reply:
I'll add that having unit test here is going to help reviewers understand what these functions do and how the behavior might change. Otherwise it's a little bit of a mystery from reading the code.

Reply:
as to the functionality, we covered this via a code review session. all of this code is unit-tested via api level tests that exorcise each of these methods in different ways. we could add unit tests to these inline at the bottom of this source as we do with our costing unit tests (that test private implementations). id rather not add to the scope of this pr though. the changes made here are all exorcised by existing (and new), albeit higher level, tests.